src/cpp/statistics/rtps/monitor-service/MonitorService.hpp

```diff
@@ -71,6 +71,8 @@ class MonitorService
 {
     static constexpr int MIN_TIME_BETWEEN_PUBS_MS = 500;
 
+    friend class MonitorServiceListener;
+
```

Comment:
Instead of making `MonitorServiceListener` a friend, add the necessary methods to `MonitorService`, and call them from `MonitorServiceListener`.

Reply:
What I meant here was to move the implementation of `on_incompatible_qos_matching` and `on_remote_proxy_data_removed` to `MonitorService`, and make the listener call those ones